packages/core/parcel-bundler/src/assets/CSSAsset.js

```diff
@@ -93,21 +100,37 @@ class CSSAsset extends Asset {
     });
   }
 
+  async pretransform() {
+    if (this.options.sourceMaps) {
+      this.contentsSourceMap = await loadSourceMap(this);
+    }
+  }
+
   async transform() {
     await postcssTransform(this);
   }
 
   getCSSAst() {
     // Converts the ast to a CSS ast if needed, so we can apply postcss transforms.
     if (!(this.ast instanceof CSSAst)) {
-      this.ast = CSSAsset.prototype.parse.call(this, this.ast.render());
+      this.ast = CSSAsset.prototype.parse.call(
+        this,
+        this.ast.render(this.name)
+      );
     }
 
     return this.ast.root;
   }
 
-  generate() {
-    let css = this.ast ? this.ast.render() : this.contents;
+  async generate() {
+    let css;
+    if (this.ast) {
+      let result = this.ast.render(this.name);
+      css = result.css;
+      if (result.map) this.sourceMap = result.map;
```

Comment:
perhaps change this to `result.map.toJSON` with the lineupdates you do later.
So this is outputs a standardised sourcemap?